File: drills/ts/harnesses/nc15_permutation_in_string.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.checkInclusion !== "function") throw new Error("__signature_mismatch__");
  return [
    ["checkInclusion('ab', 'eidbaooo')", show(true), show(solution.checkInclusion("ab", "eidbaooo"))],
    ["checkInclusion('ab', 'eidboaoo')", show(false), show(solution.checkInclusion("ab", "eidboaoo"))],
    ["checkInclusion('adc', 'dcda')", show(true), show(solution.checkInclusion("adc", "dcda"))],
  ];
}

File: drills/go/harnesses/nc15_permutation_in_string.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("checkInclusion('ab', 'eidbaooo')", true, checkInclusion("ab", "eidbaooo")),
			tc("checkInclusion('ab', 'eidboaoo')", false, checkInclusion("ab", "eidboaoo")),
			tc("checkInclusion('abc', 'ab')", false, checkInclusion("abc", "ab")),
			tc("checkInclusion('a', 'a')", true, checkInclusion("a", "a")),
		}
	})
}
